packages/next-app/src/app/daoadmin/action/[actionId]/(components)/ActionAttributeContent.tsx

```diff
@@ -1,37 +1,83 @@
 "use client";
 
 import { useRouter } from "next/navigation";
-import { HTMLProps, useContext } from "react";
 import * as React from "react";
 import { useForm } from "react-hook-form";
+import { useNetwork } from "wagmi";
 
 import { Button } from "#/components";
-import { AdminToolsContext, toSlug } from "#/contexts/AdminToolsContext";
+import { toCammelCase, useAdminTools } from "#/contexts/AdminToolsContext";
+import { fetchExistingPool } from "#/utils/fetcher";
 import { truncateAddress } from "#/utils/truncateAddress";
 
+const Input = React.forwardRef<
+  HTMLInputElement,
+  React.HTMLProps<HTMLInputElement>
+>(({ label, ...rest }: React.HTMLProps<HTMLInputElement>, ref) => {
+  return (
+    <div>
+      <label className="mb-2 block text-sm text-gray-400">{label}</label>
+      <input
+        ref={ref}
+        {...rest}
+        className="selection:color-white box-border inline-flex h-[35px] w-full appearance-none items-center justify-center rounded-[4px] bg-blackA5 px-[10px] text-[15px] leading-none text-white shadow-[0_0_0_1px] shadow-blackA9 outline-none selection:bg-blackA9 hover:shadow-[0_0_0_1px_black] focus:shadow-[0_0_0_2px_black] disabled:bg-blackA9"
+      />
+    </div>
+  );
+});
+
+const CheckPoolID = ({
+  fieldName,
+  poolName,
+  poolError,
+}: {
+  fieldName: string;
+  poolName: string | undefined;
+  poolError: string | undefined;
+}) => {
+  return (
+    <div className="mt-2 flex gap-1 text-sm text-gray-400">
+      {fieldName === "Pool ID" && poolName ? (
+        <>
+          <h1 className="text-white">Pool Name:</h1>
+          <h1>{poolName}</h1>
+        </>
+      ) : fieldName === "Pool ID" && poolError ? (
+        <h1>{poolError}</h1>
+      ) : (
+        <></>
+      )}
+    </div>
```

Comment:
This generated more lines of code to read than if it was just part of the component, I'm afraid 😬